vault/resource_jwt_auth_backend_test.go

```diff
@@ -72,6 +72,52 @@ func TestAccJWTAuthBackend_OIDC(t *testing.T) {
 	})
 }
 
+// The random numbers on the provider_config are a hash of the object.
+func TestAccJWTAuthBackend_OIDC_Provider_ConfigAzure(t *testing.T) {
+	path := acctest.RandomWithPrefix("oidc")
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testProviders,
+		CheckDestroy: testJWTAuthBackend_Destroyed(path),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccJWTAuthBackendConfigOIDCProviderConfigAzure(path),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("vault_jwt_auth_backend.azure", "oidc_discovery_url", "https://accounts.google.com"),
+					resource.TestCheckResourceAttr("vault_jwt_auth_backend.azure", "provider_config.#", "1"),
+					resource.TestCheckResourceAttr("vault_jwt_auth_backend.azure", "provider_config.2995719985.provider", "azure"),
+				),
+			},
+		},
+	})
+}
+
+// The random numbers on the provider_config are a hash of the object.
```

Comment:
Is it worth leaving a quick instruction on how to regenerate the new hash if someone needs to change the object under test?

Reply:
Actually, I'm going to add a function to calculate it because I had to do some hacky stuff to get the answer.

Reply:
Added a hashing function in.

Reply:
Unfortunately there were issues with the hashing function, so I reverted this. I'll make a note internally how I got this hash. With the Terraform v2 SDK this is no longer needed so once we upgrade we can clean this up.